### ISSUE TITLE
Mercurial fails to run "Empty hg version output:"

### ISSUE THREAD

**Issue by mitchhentges** [flathub/com.jetbrains.PyCharm-Professional#22]
### TL;DR

The Python bundled with PyCharm doesn't check the user directory for packages to import.

### Details

When the path to `hg` is provided in "Settings > Version Control > Mercurial > Path to Mercurial executable", clicking the "Test" button gives the error "Empty hg version output":.
Additionally, there's an error shown in the event log:
```
5:11 p.m.	Could not execute hg status command : [Traceback (most recent call last):,   File "/var/run/host/usr/local/bin/hg", line 24, in <module>,     from hgdemandimport import tracing, ModuleNotFoundError: No module named 'hgdemandimport']
```

This can be further investigated from the built-in terminal, by attempting to run `hg` from there:
```
sh-5.0$ /home/mitch/.local/bin/hg --version
Traceback (most recent call last):
  File "/home/mitch/.local/bin/hg", line 24, in <module>
    from hgdemandimport import tracing
ModuleNotFoundError: No module named 'hgdemandimport'
```

Outside of PyCharm, if I look for the location of `hgdemandimport`, I see that it's in my user "site-packages":
```
(host) $ python3 -c "import hgdemandimport; print(hgdemandimport)"
<module 'hgdemandimport' from '/home/mitch/.local/lib/python3.7/site-packages/hgdemandimport/__init__.py'>
```

Within PyCharm's terminal, I see (of course) that the same command fails:
```
sh-5.0$ python3 -c "import hgdemandimport; print(hgdemandimport)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'hgdemandimport'
```

Python can't seem to find the package. Let's check the `sys.path`:
```
sh-5.0$ python3 -c "import sys; print(sys.path)"
['', '/usr/lib/python37.zip', '/usr/lib/python3.7', '/usr/lib/python3.7/lib-dynload', '/usr/lib/python3.7/site-packages', '/app/lib/python3.7/site-packages']
```

Aha! Python can't find `hgdemandimport` because its location (`/home/mitch/.local/lib/python3.7/site-packages/`) is not in the `sys.path`!

### Solutions

* We can bundle `hg` with the flatpak package, but then it can get out-of-date, and users' extensions may still have issues loading if they have dependencies.
* We can add `$HOME/.local/lib/<bundled-python-version>/site-packages` to the python path
* [According to this other PR](https://github.com/flathub/com.jetbrains.PyCharm-Community/pull/99), updating the runtime may help with python packages? Unsure.
* (Other solution that hopefully you'll think of :smile: )

**Comment by johanfleury** [comment]
Does this work on PyCharm Community ?

> According to this other PR, updating the runtime may help with python packages? Unsure.

I’ve created #20 to track this a few weeks ago. Unfortunately I’m not confident enough with Flatpak to be sure I won’t break anything and my free time is limited. Also, I mainly packaged PyCharm Professional because a colleague wanted it, but I’m not really using it myself.

If you want to have a look at this, your contribution is more than welcome :)

**Comment by mitchhentges** [comment]
Wow, this ticket was from last November?! Sorry for the delayed response, hah :)
This doesn't work with PyCharm Community either, so my theory on updating the runtime turns out not to be fruitful.

Also, while I was digging into this, my use case involved using multiple versions of Python. However, when I pointed the flatpak towards Python versions I had set up using `pyenv`, I ran into `GLIBC` version mismatch errors. That makes sense, since the sandbox isn't guaranteed to align with my host OS, and I'm not sure how to rectify that class of problem.

In the meantime, I've worked around this by extracting the official `tar.gz` into `~/.local/opt` and setting up a `.desktop` file. Since PyCharm auto-updates, it works for me. Unfortunately, I've got no good ideas for improving the flatpak situation just yet.

I'll leave this ticket open since _technically_ it's still a problem, but feel free to close it if you'd prefer.

> Also, I mainly packaged PyCharm Professional because a colleague wanted it, but I’m not really using it myself.

That's some dedication right there! Thanks for making this package, and sorry I couldn't be more help :)

**Comment by freeseacher** [comment]
As far as i can see the best way to provide access to mercurial is to add extension with it. 
See https://github.com/freeseacher/com.jetbrains.PyCharm.Extensions.Python310 for example
